app/controllers/onboarding/telegram_controller.rb

```diff
@@ -1,87 +1,46 @@
 # frozen_string_literal: true
 
-module Onboarding
-  class TelegramController < ApplicationController
-    include JwtHelper
-
-    skip_before_action :require_login
-    before_action -> { verify_onboarding_jwt(jwt_param) }, except: %i[update]
-    before_action :verify_telegram_authentication_and_integrity, only: :create
+require 'securerandom'
 
-    layout 'onboarding'
+module Onboarding
+  class TelegramController < ChannelController
+    skip_before_action :verify_jwt, only: %i[link fallback]
+    before_action :ensure_contributor_exists, only: %i[link fallback]
 
     def show
-      @jwt = jwt_param
+      super
+      @contributor.telegram_onboarding_token = SecureRandom.alphanumeric(8)
```

Comment:
I think for UX reasons, we should either use upper- or lowercase characters only and make sure to handle onboarding tokens case-insensitively.